storage/client/src/main/java/com/azure/storage/file/FileAsyncClient.java

```diff
@@ -3,77 +3,392 @@
+                            .url(azureFileStorageClient.url())
+                            .version(azureFileStorageClient.version())
+                            .build();
+    }
+
+    /**
+     * Constructor of FileAsyncClient
+     * @param endpoint
+     * @param httpPipeline
+     * @param shareName
+     * @param filePath
+     */
+    FileAsyncClient(URL endpoint, HttpPipeline httpPipeline, String shareName, String filePath, String shareSnapshot) {
+        this.shareName = shareName;
+        this.filePath = filePath;
+        this.shareSnapshot = shareSnapshot;
+        this.azureFileStorageClient = new AzureFileStorageBuilder().pipeline(httpPipeline)
+                          .url(endpoint.toString())
+                          .build();
+    }
+
+    /**
+     * Get URL from azureFileStorageClient.
+     * @return
+     */
+    public URL url() throws Exception {
+        return new URL(azureFileStorageClient.url());
+    }
+
+    /**
+     * Builder of FileAsyncClient and FileClient.
+     * @return
+     */
+    public static FileClientBuilder builder() {
+        return new FileClientBuilder();
     }
 
-    public static FileClientBuilder asyncBuilder() {
-        throw new UnsupportedOperationException();
+    /**
+     * Create a new file in storage.
+     * @param maxSize
+     * @return
+     */
+    public Mono<Response<FileInfo>> create(long maxSize) {
+        return create(maxSize, null, null);
     }
 
+    /**
+     * Create a new file in storage.
+     * @param maxSize
+     * @param httpHeaders
+     * @param metadata
+     * @return
+     */
     public Mono<Response<FileInfo>> create(long maxSize, FileHTTPHeaders httpHeaders, Map<String, String> metadata) {
-        throw new UnsupportedOperationException();
+        return azureFileStorageClient.files().createWithRestResponseAsync(shareName, filePath, maxSize, null, metadata, httpHeaders, Context.NONE)
+            .map(this::createResponse);
     }
 
+    /**
+     * Copy from another source.
+     * @param sourceUrl
+     * @param metadata
+     * @return
+     */
     public Mono<Response<FileCopyInfo>> startCopy(String sourceUrl, Map<String, String> metadata) {
-        throw new UnsupportedOperationException();
+        return azureFileStorageClient.files().startCopyWithRestResponseAsync(shareName, filePath, sourceUrl, null, metadata, Context.NONE)
+                    .map(this::startCopyResponse);
     }
 
+    /**
+     * Abort the copy.
+     * @param copyId
+     * @return
+     */
     public Mono<VoidResponse> abortCopy(String copyId) {
-        throw new UnsupportedOperationException();
+        return azureFileStorageClient.files().abortCopyWithRestResponseAsync(shareName, filePath, copyId, Context.NONE)
+                    .map(VoidResponse::new);
+    }
+    /**
+     * Download with properties
+     * @return
+     */
+    public Mono<Response<FileDownloadInfo>> downloadWithProperties() {
```

Comment:
Why did we name this downloadWithProperties instead of just download? All other languages are calling it download.